### PR TITLE
Persistence/login/signup view

### LINKED ISSUE #28: Implement login/signup use case in for User
Login and signup use case with persistence across program runs, saving user login data.
Both use cases need a view, with login being the default and having a signup option to switch to the signup view.
Once successfully logged in, it should switch to our recipe search view

### PR DESCRIPTION
**Pull request by RqedK**
Implemented the view for both the sign_up and the log_in use cases 
Fixed the implementation of the CSVDataAccessObject (csv file is now created to store user information)

Functionality:
Default view: Login (option to sign up button included which switches to the signup view)
once successfully logged in, switches to recipe search 
login feature has data persistence across program runs

### CODE REVIEW

File: src/main/java/app/Config.java [hunk 1]
```diff
@@ -9,8 +8,6 @@
 import frameworks.gui.GUI;
 import frameworks.gui.SwingGUI;
 
-// UseCaseFactory
```
Comment (hhpfiona) [inline src/main/java/app/Config.java:14]: pretty sure we need UseCaseFactory just per csc207 requirements

File: src/main/java/app/Main.java [hunk 2]
```diff
@@ -1,20 +1,10 @@
 package app;
+
 import frameworks.gui.GUI;
-import java.util.Scanner;
 
 public class Main {
     public static void main(String[] args) {
-        // Check if the user wants to run in console mode or GUI mode
```
Comment (hhpfiona) [inline src/main/java/app/Main.java:7]: i think this is a good choice

File: src/main/java/use_cases/log_in/interface_adapter/controller/LoginController.java [hunk 3]
```diff
@@ -3,13 +3,31 @@
 import use_cases.log_in.use_case.input_data.LoginInputBoundary;
 import use_cases.log_in.use_case.input_data.LoginInputData;
 
+/**
```
Comment (hhpfiona) [inline src/main/java/use_cases/log_in/interface_adapter/controller/LoginController.java:6]: nice documentations overall for this file

**Review by hhpfiona — APPROVED** [review]

File: src/main/java/app/Config.java (hunk 1 above)
Comment (parkhangorodsky) [reply src/main/java/app/Config.java:14]: agree with this!

**Review by parkhangorodsky — CHANGES_REQUESTED** [review]
A few requests:
1) merge the current main to your brain - it doesn't seem like it has my merge for NutritionDisplay
2) also get the image files back that you removed - we'll still use those as of now.
3) undelete the UseCaseFactory (since we need it for the structure)

**Review by parkhangorodsky — CHANGES_REQUESTED** [review]
i think just remove the target files from the PR and you're good

**Review by parkhangorodsky — APPROVED** [review]
looks good - the decision is to merge this as is, then make another branch (and PR) to fix the issues with the target files - since it's not related to the contents of this PR

**Review by hhpfiona — APPROVED** [review]
*paula's nutrition print functionality. 

this looks okay but we would need new branch to adjust issues with target

**Pull request merged**